students/itsabhi/Lesson6/mailroom4v1.py

```diff
@@ -0,0 +1,99 @@
+#! /usr/bin/env python3
+
+
+def list_of_donors(donor_hist):
+    donor_list = []
+    for k in donor_hist:
+        donor_list.append(k)
+    return (donor_list)
+
+
+def sending_thanksv2(donor_hist):
+    # If already exists, add donation amt to history, else update records as a new donor & then print thank you letter.
+    donor = input("Type the Full Name of the donor here or type List to display current donors : ")
+    if donor == "list":
+                for k in donor_hist:
+                    print(k)
+                donor = input("Type the Full Name of the donor here: ")
+    if donor in donor_hist.keys():
+        while True:
+            amount = input("Enter Donation value: ")
+            donation_hist = donor_hist.get(donor)
+            try:
+                donor_hist[donor] = (donation_hist + [int(amount)])
+            except ValueError:
+                print("Invalid donation entered..")
+            else:
+                break
+    else:
+        while True:
+            amount = input("Enter Donation value: ")
+            try:
+                donor_hist.update({donor: [int(amount)]})  #Convert donation value from string to integer
+            except ValueError:
+                print("Invalid donation entered..")
+            else:
+                break
```

Comment:
Can you think of a way to make to avoid the double `while True` loops and make this section more DRY?